pandas/tools/tests/test_concat.py

```diff
@@ -2151,6 +2151,25 @@ def test_concat_multiindex_rangeindex(self):
         exp = df.iloc[[2, 3, 4, 5], :]
         tm.assert_frame_equal(res, exp)
 
+    def test_concat_multiindex_dfs_with_deepcopy(self):
+        # GH 9967
+        from copy import deepcopy
+        example_multiindex1 = pd.MultiIndex.from_product([['a'], ['b']])
+        example_dataframe1 = pd.DataFrame([0], index=example_multiindex1)
+
+        example_multiindex2 = pd.MultiIndex.from_product([['a'], ['c']])
+        example_dataframe2 = pd.DataFrame([1], index=example_multiindex2)
+
+        example_dict = {'s1': example_dataframe1, 's2': example_dataframe2}
+        result = pd.concat(deepcopy(example_dict), names=['testname'])
+        expected_index = pd.MultiIndex(levels=[['s1', 's2'],
+                                               ['a'],
+                                               ['b', 'c']],
+                                       labels=[[0, 1], [0, 0], [0, 1]],
+                                       names=['testname', None, None])
+        expected = pd.DataFrame(np.array([[0], [1]]), index=expected_index)
```

Comment:
on windows, ``np.array([[0], [1]])`` is ``int32``, so pass a ``dtype='int64'`` here on the expected, or
just use ``[[0], [1]]`` which always returns ``int64``

```
======================================================================
FAIL: test_concat_multiindex_dfs_with_deepcopy (pandas.tools.tests.test_concat.TestConcatenate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\pandas\lib\site-packages\pandas\tools\tests\test_concat.py", line 2171, in test_concat_multiindex_dfs_with_deepcopy
    tm.assert_frame_equal(result, expected)
  File "C:\Python27_64\envs\pandas\lib\site-packages\pandas\util\testing.py", line 1313, in assert_frame_equal
    obj='DataFrame.iloc[:, {0}]'.format(i))
  File "C:\Python27_64\envs\pandas\lib\site-packages\pandas\util\testing.py", line 1154, in assert_series_equal
    assert_attr_equal('dtype', left, right)
  File "C:\Python27_64\envs\pandas\lib\site-packages\pandas\util\testing.py", line 878, in assert_attr_equal
    left_attr, right_attr)
  File "C:\Python27_64\envs\pandas\lib\site-packages\pandas\util\testing.py", line 1018, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: Attributes are different
Attribute "dtype" are different
[left]:  int64
[right]: int32
----------------------------------------------------------------------
```